lib/jsonschema.lua

```diff
@@ -763,9 +775,20 @@ generate_validator = function(ctx, schema)
     end
 
     if schema.uniqueItems then
-      ctx:stmt(sformat('  local ok, item1, item2 = %s(%s)', ctx:libfunc('lib.unique_item_in_array'), ctx:param(1)))
-      ctx:stmt(sformat('  if not ok then', ctx:libfunc('lib.unique_item_in_array'), ctx:param(1)))
-      ctx:stmt(sformat('    return false, %s("expected unique items but items %%d and %%d are equal", item1, item2)', ctx:libfunc('string.format')))
+      ctx:stmt(sformat('  local is_simple_tab = %s(%s)', ctx:libfunc('lib.is_simple_table'), ctx:param(1)))
+      ctx:stmt(sformat('  if is_simple_tab then'))
+      ctx:stmt(sformat('    local ok, item1, item2 = %s(%s)', ctx:libfunc('lib.unique_item_in_array'), ctx:param(1)))
+      ctx:stmt(sformat('    if not ok then', ctx:libfunc('lib.unique_item_in_array'), ctx:param(1)))
+      ctx:stmt(sformat('      return false, %s("expected unique items but items %%d and %%d are equal", item1, item2)', ctx:libfunc('string.format')))
+      ctx:stmt(        '    end\n')
+      ctx:stmt(        '  else')
+      ctx:stmt(sformat('    for i=2, #%s do', ctx:param(1)))
+      ctx:stmt(        '      for j=1, i-1 do')
+      ctx:stmt(sformat('        if %s(%s[i], %s[j]) then', ctx:libfunc('lib.deepeq'), ctx:param(1), ctx:param(1)))
+      ctx:stmt(sformat('          return false, %s("expected unique items but items %%d and %%d are equal", i, j)', ctx:libfunc('string.format')))
```

Comment:
Variable `i` is greater than `j`, can we adjust the hint messsage?

Reply:
you are right, fix it later